File: DacSanMienTrung_Backend/src/main/java/com/example/dacsanmientrung_backend/service/AuthService.java
package com.example.dacsanmientrung_backend.service;

import com.example.dacsanmientrung_backend.dto.request.LoginRequest;
import com.example.dacsanmientrung_backend.dto.request.RegisterRequest;
import com.example.dacsanmientrung_backend.dto.response.LoginResponse;
import com.example.dacsanmientrung_backend.dto.response.UserResponse;

public interface AuthService {

    UserResponse register(RegisterRequest request);

    LoginResponse login(LoginRequest request);

    UserResponse getUserById(Integer id);

    UserResponse getUserByEmail(String email);
}

File: DacSanMienTrung_FE/src/services/authService.js
import { getApi, postApi } from './apiClient'

export const login = async (data) => {
  return postApi('/auth/login', data)
}

export const register = async (data) => {
  return postApi('/auth/register', data)
}

export const getUserById = async (id) => {
  return getApi(`/auth/users/${id}`)
}

export const getUserByEmail = async (email) => {
  return getApi(`/auth/users/email?email=${encodeURIComponent(email)}`)
}
